tests/xtensor/test_shape.py

```diff
@@ -102,3 +102,59 @@ def test_multiple_stacks():
     res = fn(x_test)
     expected_res = x_test.stack(new_dim1=("a", "b"), new_dim2=("c", "d"))
     xr_assert_allclose(res[0], expected_res)
+
+
+def test_unstack():
+    unstacked_dims = {"a": 2, "b": 3, "c": 5, "d": 7}
+    dims = ("abcd",)
+    x = xtensor("x", dims=dims, shape=(2 * 3 * 5 * 7,))
+    outs = [
+        unstack(
+            x,
+            abcd=(
+                {d: l for d, l in unstacked_dims.items() if d in dims_to_unstack}
+                | (
+                    {}
+                    if set(dims_to_unstack) == set(unstacked_dims)
+                    else {
+                        "other": int(
+                            np.prod(
+                                [
+                                    l
+                                    for d, l in unstacked_dims.items()
+                                    if d not in dims_to_unstack
+                                ]
+                            )
+                        )
+                    }
+                )
+            ),
+        )
```

Comment:
This is a bit hard for me to read

Reply:
I think I get what you were trying to do with the test (60% confidence), and I think it has no parallel to what our poor-mans unstack can do. We basically can only unstack "consecutive dimensions", whereas xarray will always know what a bunch of stacked dimensions correspond to, and can unstack "non-consecutive" dimensions.

I think for our purposes we want to always get an identity if we do `transpose(unstack(stack(new_dim=stacked_dims), new_dim=original_stacked_dims), original_dims)`, where`original_stacked_dims` contains the same dims, in the same order and with the same sizes.

I added a test more like that, that maybe we can parametrize with the powerset approach?